File: Kolmogorov_entropy_of_a_process_or_series.cpp
#include <iostream>
#include <cmath>

double kolmogorovEntropy(int* outcomes, double* probabilities, int n) {
    double entropy = 0.0;
    for (int i = 0; i < n; i++) {
        entropy += -probabilities[i] * log2(probabilities[i]);
    }
    return entropy;
}

int main() {
    // Test the function
    int outcomes[] = {1, 2, 3, 4};
    double probabilities[] = {0.25, 0.25, 0.25, 0.25};
    std::cout << kolmogorovEntropy(outcomes, probabilities, 4) << std::endl;
    return 0;
}

File: Kolmogorov_entropy_of_a_process_or_series.rs
use std::f64;

fn kolmogorov_entropy(outcomes: &[f64], probabilities: &[f64]) -> f64 {
    let mut entropy = 0.0;
    for (outcome, probability) in outcomes.iter().zip(probabilities) {
        entropy += -probability * probability.log2();
    }
    entropy
}

// Test the function
let outcomes = [1.0, 2.0, 3.0, 4.0];
let probabilities = [0.25, 0.25, 0.25, 0.25];
println!("{}", kolmogorov_entropy(&outcomes, &probabilities));
